File: spec/cypress/integration/messenger/translations_spec.js

import {
  translations,
  openMessenger
} from '../../support'

describe('Translations Spec', function () {
  it('user lang en', function () {
    cy.appScenario('basic')
    translations()
    openMessenger(($body) => {
      expect($body.html()).to.contain('Start a conversation')
      expect($body.html()).to.contain('hello friend')
      expect($body.html()).to.contain('we are here to help')
      cy.appEval('App.last.app_users.size').then((res) => {
        expect(res).to.equal(1)
      })
      cy.appEval('App.last.app_users.last.lang').then((res) => {
        expect(res).to.equal('en')
      })
    }, { params: { lang: 'en' } })
  })

  it('user lang es', function () {
    cy.appScenario('basic')
    translations()
    openMessenger(($body) => {
      expect($body.html()).to.contain('Inicia una conversación')
      /* expect($body.html()).to.contain('hola')
      expect($body.html()).to.contain('estamos aqui')
      expect($body.html()).to.contain('somos un equipo') */
      cy.appEval('App.last.app_users.size').then((res) => {
        expect(res).to.equal(1)
      })

      cy.appEval('App.last.app_users.last.lang').then((res) => {
        expect(res).to.equal('es')
      })
      cy.appEval('App.last.app_users.last.name').then((res) => {
        expect(res).to.equal('miguel')
      })
    }, { params: { lang: 'es' } })
  })

  it('visitor lang en', function () {
    cy.appScenario('basic')
    translations()
    openMessenger(($body) => {
      expect($body.html()).to.contain('Start a conversation')
      expect($body.html()).to.contain('hello friend')
      cy.appEval('App.last.app_users.size').then((res) => {
        expect(res).to.equal(1)
      })
    }, { params: { lang: 'en' }, headers: {} }, true)
  })

  it('visitor lang es property', function () {
    cy.appScenario('basic')
    translations()
    openMessenger(($body) => {
      expect($body.html()).to.contain('Inicia una conversación')
      expect($body.html()).to.contain('Hola')
      expect($body.html()).to.contain('somos un equipo')
      cy.appEval('App.last.app_users.size').then((res) => {
        expect(res).to.equal(1)
      })

      cy.appEval('Visitor.last.lang').then((res) => {
        expect(res).to.equal('es')
      })
    }, { params: { lang: 'es' }, headers: {} }, true)
  })

  it('lang en', function () {
    cy.appScenario('start_conversation_from_agent')

    cy.wait(2000)

    openMessenger(($body) => {
      // cy.wrap($body.find("h2")[1]).contains('Start a conversation')
      expect($body.html()).to.contain('Start a conversation')
      cy.wrap($body).contains('a few seconds ago')
      /* cy.wrap($body).xpath('/html/body/div/div/div/div[2]/div/div[1]/div[2]/div')
        .click().then(() => {
          expect($body.html()).to.contain('foobar')
          cy.wrap($body)
            .xpath('/html/body/div/div/div/div[2]/div/div/div/div[2]/div/div/textarea')
            .type('oeoe \n').then(() => {
              cy.wait(3000)
              cy.wrap($body).contains('oeoe')
            })
        }) */
    }, { params: { lang: 'en' } }, false)
  })

  it('lang es', function () {
    cy.appScenario('start_conversation_from_agent')
    openMessenger(($body) => {
      expect($body.html()).to.contain('Inicia una conversación')
      // cy.wrap($body).contains('a few seconds ago')
      /* cy.wrap($body).xpath('/html/body/div/div/div/div[2]/div/div[1]/div[2]/div')
        .click().then(() => {
          expect($body.html()).to.contain('foobar')
          cy.wrap($body)
            .xpath('/html/body/div/div/div/div[2]/div/div/div/div[2]/div/div/textarea')
            .type('oeoe \n').then(() => {
              cy.wrap($body).contains('oeoe')
            })
        }) */
    }, { params: { lang: 'es' } }, false)
  })
})

File: spec/system/messenger/translations_spec.rb
require "rails_helper"

RSpec.describe "Translations Spec", type: :system, js: true do
  let!(:app) do
    Plugin.restore_plugins_from_fs

    FactoryBot.create(:app, encryption_key: "unodostrescuatro",
                            active_messenger: "true",
                            state: "enabled",
                            domain_url: "*")
  end

  let!(:agent) do
    agent = app.add_agent({
                            email: "test@test.cl",
                            name: "sharleena",
                            password: "123456",
                            password_confirmation: "123456"
                          },
                          role_attrs: {
                            role: "admin"
                          })
    agent.agent
  end

  def translations
    app.update({
                 greetings_es: "Hola amigo",
                 greetings_en: "hello friend",

                 intro_en: "we are here to help",
                 intro_es: "somos un equipo genial",

                 tagline_es: "estamos aqui para ayudarte",
                 tagline_en: "we are an awesome team"
               })
  end

  before :each do
    translations
  end

  it "user lang en" do
    open_new_messenger({ params: { lang: "en" } }) do |page|
      expect(page).to have_content("Start a conversation")
      expect(page).to have_content("hello friend")
      expect(page).to have_content("we are here to help")

      # Verifying app user properties
      expect(app.app_users.size).to eq(1)
      expect(app.app_users.last.lang).to eq("en")
    end
  end

  it "user lang es" do
    open_new_messenger({ params: { lang: "es" } }) do
      expect(page).to have_content("Inicia una conversaci\u00F3n")
      expect(app.app_users.size).to eq(1)
      expect(app.app_users.last.lang).to eq("es")
      expect(app.app_users.last.name).to eq("miguel")
    end
  end

  it "visitor lang en" do
    open_new_messenger({ params: { lang: "en" } }, sessionless: true) do
      expect(page).to have_content("Start a conversation")
      expect(page).to have_content("hello friend")
      expect(app.app_users.size).to eq(1)
    end
  end

  it "visitor lang es property" do
    open_new_messenger({ params: { lang: "es" } }, sessionless: true) do
      expect(page).to have_content("Inicia una conversaci\u00F3n")
      expect(page).to have_content("Hola")
      expect(app.app_users.size).to eq(1)
      expect(Visitor.last.lang).to eq("es")
    end
  end

  it "lang en start conversation" do
    # app.add_user(email: "test@test.cl")

    # start_conversation_from_agent

    open_new_messenger({ params: { lang: "en" } }, sessionless: true) do
      expect(page).to have_content("Start a conversation")
      expect(page).to have_content("hello")
      expect(app.app_users.size).to eq(1)

      start_conversation_from_agent

      sleep(2)
      # expect(Visitor.last.lang).to eq('en')
      # click_link("See previous")

      # sleep(2)
      expect(page).to have_content("foobar")
    end
  end

  # Additional test cases...
end
